sdk/src/transaction_context.rs

```diff
@@ -859,14 +869,30 @@ impl<'a> BorrowedAccount<'a> {
         {
             return Ok(());
         }
+        let old_length = self.get_data().len();
         // Only the owner can change the length of the data
-        if new_length != self.get_data().len() && !self.is_owned_by_current_program() {
+        if new_length != old_length && !self.is_owned_by_current_program() {
             return Err(InstructionError::AccountDataSizeChanged);
         }
         // The new length can not exceed the maximum permitted length
         if new_length > MAX_PERMITTED_DATA_LENGTH as usize {
             return Err(InstructionError::InvalidRealloc);
         }
+        if self
+            .transaction_context
+            .is_cap_accounts_data_allocations_per_transaction_enabled
+        {
+            // The resize can not exceed the per-transaction maximum
+            let length_delta = (new_length as i64).saturating_sub(old_length as i64);
+            if self
+                .transaction_context
+                .accounts_resize_delta()?
+                .saturating_add(length_delta)
+                > MAX_PERMITTED_ACCOUNTS_RESIZE_DELTA
+            {
+                return Err(InstructionError::MaxAccountsDataSizeExceeded);
```

Comment:
This instruction error should probably be renamed to indicate that the resize delta limit of a single transaction was exceeded. This reads as if the cluster wide account data size limit was exceeded.

Reply:
That's fine with me